subsys/bluetooth/controller/Kconfig.ll_sw_split

```diff
@@ -598,12 +598,44 @@ config BT_CTLR_DYNAMIC_INTERRUPTS
 	  permit use of SoC's peripheral for custom use when Bluetooth is not
 	  enabled.
 
+choice
+	prompt "Optimization options"
+	depends on !LTO
+	default BT_CTLR_OPTIMIZE_FOR_SPEED
+
+config BT_CTLR_OPTIMIZE_FOR_APP_DEFAULT
+	bool "Optimize for application defaults"
+	help
+	  Optimize compilation of Controller for application's default.
+
+config BT_CTLR_OPTIMIZE_FOR_SIZE
+	bool "Optimize for space"
+	help
+	  Optimize compilation of Controller for code size.
+
+	  This is the Zephyr project default, but is overridden to use optimize
+	  for speed when compiling the Controller to reduce application's ISR
+	  and thread latencies.
+
 config BT_CTLR_OPTIMIZE_FOR_SPEED
-	prompt "Optimize for Speed" if !(SOC_SERIES_NRF51X && BT_CTLR_LE_ENC)
-	bool
-	default y if BT_CTLR_LE_ENC
+	bool "Optimize for Speed"
+	help
+	  Optimize compilation of Controller for execution speed. As the
+	  Controller's Upper Link Layer and Lower Link Layer execute in ISRs,
+	  use of speed optimizations reduces application's ISR and thread
+	  latencies.
+
+endchoice
+
+config BT_CTLR_LTO
```

Comment:
From your commit message it looks like `BT_CTLR_OPTIMIZE_FOR_SIZE` is worse for size than `BT_CTLR_LTO`. 

Is there a downside to `BT_CTLR_LTO` since we have both? Otherwise it would appear as `BT_CTLR_OPTIMIZE_FOR_SIZE` could be removed as just being a worse version of `BT_CTLR_LTO`. 

What does the size comparison look like if you just compared `BT_CTLR_OPTIMIZE_FOR_SIZE` to `LTO`, i.e. have the same values for `SHARED_INTERRUPTS` and `ISR_TABLES_LOCAL_DECLARATION`?

Reply:
LTO option for gcc compiler does not optimize when generating object files (my understanding for now), the linker finds instruction patterns to reuse code, hence optimization considers across object files (copy-paste code across c files sometimes lead to lower code space with LTO!)

LTO gives ~4% more optimization compared to `-Os` (compiler optimization for size) in the Controller code base.

Compiler optimization for Size ver Speed gives ~10%.

For:
```
-CONFIG_BT_CTLR_LTO=y
+CONFIG_BT_CTLR_OPTIMIZE_FOR_SIZE=y
+CONFIG_SHARED_INTERRUPTS=y
+CONFIG_ISR_TABLES_LOCAL_DECLARATION=y
+# CONFIG_BT_CTLR_LTO=y
```

I get:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      233528 B       256 KB     89.08%
             RAM:       60128 B        64 KB     91.75%
           SRAM1:           0 B        64 KB      0.00%
        IDT_LIST:           0 B        32 KB      0.00%
```

i.e  more code added to `-Os` with `SHARED_INTERRUPTS` and `ISR_TABLES_LOCAL_DECLARATION`

Reply:
Thanks. Does it make sense to keep both `BT_CTLR_OPTIMIZE_FOR_SIZE` and `BT_CTLR_LTO` then? Or does `BT_CTLR_LTO` only work for GCC?

Reply:
for now ` BT_CTLR_LTO` needs `LTO` in Zephyr... but I do not see any restrictions set on use of GCC for `LTO` in Zephyr Kconfig other than `ARM` and around the interrupt table constructs.

Today  by this PR,  `BT_CTLR_LTO` uses the zephyr project default of `-Os` to give the best code reduction. I tried to force `BT_CTLR_LTO` with `-Ofast`, this resulted in 16K Flash overflow. Probably, LTO and `-Ofast`  is broken, hence, I will keep the changes in this PR for now as-is, to not mix compiler optimiaztion with link time optimization.